### PR TITLE
Now passing the URL and options as extra parameters for the middleware.

### PR DESCRIPTION
**Pull request by SanderBenschop**

### CODE REVIEW

File: tests/request.test.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ describe('requests', () => {
 
       const json = await get('api/pokemon/1');
       expect(json).toEqual({ id: 1 });
+      expect(middleware.parseJSON).toHaveBeenCalledWith(expect.any(Promise), expect.any(String), undefined);
```
Comment (Pouja) [inline tests/request.test.js]: Why the any string? It should be the `/api/pokemon/1` url right?

**Comment by MrHus** [comment]
Can you describe the use case for this change?

**Comment by SanderBenschop** [comment]
The usecase is that I want to globally catch an HTTP status, trigger a two-factor authentication popup, and afterwards repeat the request. To do this I need to know the URL and parameters.

File: tests/request.test.js (hunk 1 above)
Comment (SanderBenschop) [reply tests/request.test.js]: Yeah you're right, I'll change the test.

**Comment by MrHus** [comment]
> The usecase is that I want to globally catch an HTTP status, trigger a two-factor authentication popup, and afterwards repeat the request. To do this I need to know the URL and parameters.

I see this is a good usecase I wonder however if sending the url and parameters as two separate things is useful in the long run, if more info is needed more and more parameters will need to be added. 

Perhaps a singular "request" object which is passed on which contains all the information about the request in a single object will make the api more robust for the future.

**Comment by SanderBenschop** [comment]
Will do!

**Comment by SanderBenschop** [comment]
I made the suggested changes, can you please check it again?

**Pull request merged**